iotdb-client/cli/src/main/java/org/apache/iotdb/tool/ImportTsFile.java

```diff
@@ -0,0 +1,459 @@
+      return false;
+    }
+  }
+
+  public static File createSuccessDir(CommandLine commandLine) {
+    if (commandLine.getOptionValue(SUCCESS_DIR_ARGS) != null) {
+      successDir = commandLine.getOptionValue(SUCCESS_DIR_ARGS);
+    }
+    File file = new File(successDir);
+    if (!file.isDirectory()) {
+      file.mkdirs();
+    }
+    return file;
+  }
+
+  public static File createFailDir(CommandLine commandLine) {
+    if (commandLine.getOptionValue(FAIL_DIR_ARGS) != null) {
+      failDir = commandLine.getOptionValue(FAIL_DIR_ARGS);
+    }
+    File file = new File(failDir);
+    if (!file.isDirectory()) {
+      file.mkdirs();
+    }
+    return file;
+  }
+
+  public static int importFromTargetPath() {
+    try {
+      sessionPool = new SessionPool(host, Integer.parseInt(port), username, password, threadNum);
+      File file = new File(source);
+      sourceFullPath = file.getAbsolutePath();
+      if (!file.isFile() && !file.isDirectory()) {
+        ioTPrinter.println("File not found!");
+        return CODE_ERROR;
+      }
+      traverse(file);
+      asyncImportTsFiles();
+    } catch (InterruptedException e) {
+      ioTPrinter.println("Traversing file exceptions: " + e.getMessage());
+      return CODE_ERROR;
+    } finally {
+      if (sessionPool != null) {
+        sessionPool.close();
+      }
+    }
+    return CODE_OK;
+  }
+
+  public static void traverse(File file) throws InterruptedException {
+    if (file.isFile()) {
+      linkedBlockingQueue.put(file.getAbsolutePath());
+    } else if (file.isDirectory()) {
+      File[] files = file.listFiles();
+      if (files != null) {
+        for (File f : files) {
+          traverse(f);
+        }
+      }
+    }
+  }
+
+  public static void asyncImportTsFiles() {
+    List<Thread> list = new ArrayList<>(threadNum);
+    for (int i = 0; i < threadNum; i++) {
+      Thread thread = new Thread(ImportTsFile::importTsFile);
+      thread.start();
+      list.add(thread);
+    }
+    list.forEach(
+        thread -> {
+          try {
+            thread.join();
+          } catch (InterruptedException e) {
+            ioTPrinter.println("importTsFile thread join interrupted: " + e.getMessage());
+          }
+        });
+  }
+
+  public static void importTsFile() {
+    String filePath;
+    try {
+      while ((filePath = linkedBlockingQueue.poll()) != null) {
+        String sql = "load '" + filePath + "' onSuccess=none ";
+        try {
+          sessionPool.executeNonQueryStatement(sql);
+          ioTPrinter.println("Import [ " + filePath + " ] file success");
+          processingSuccess(filePath);
+        } catch (Exception e) {
+          ioTPrinter.println("Import [ " + filePath + " ] file fail: " + e.getMessage());
+          processingFailed(filePath);
+        }
+      }
+    } catch (Exception e) {
+      ioTPrinter.println("Import file exceptions: " + e.getMessage());
+    }
+  }
+
+  public static void processingSuccess(String filePath) {
```

Comment:
Maybe avoiding using "Paths.get" each time by pass in the transformed path directly?